### ISSUE TITLE
Sharing

### ISSUE THREAD

**Issue by julianpoy** [FeyNote/FeyNote#75]
Allow users to share their content.

- [x] Allow users to share a single artifact by URL to users who are not logged in as read-only
- [x] Allow users to share a single artifact with other platform users as read/write for collaboration
- [ ] Allow users to share a collection/label with other platform users as read/write for collaboration
- [ ] Allow users to share a collection/label by URL to users who are not logged in as read-only
- [ ] Support a "shared with me" list, displaying whatever was shared
  - [ ] If a collection was shared, display that
  - [ ] If an artifact was shared, display that
- [ ] Support a "add to my library" button, which will add it to their native collection list, but keep it in "shared with me" (and does not duplicate the artifact)
- [x] Search should be supported on the "shared with me" list

Some thoughts:
- [ ] It might be nice to have everything just listed on the dashboard, with a "shared with me" category rather than a separate page. This would eliminate a lot of headache.
- [x] We need a way for the user to remove things shared to them. This would probably replace the "delete" button on any of these shared items, if they don't own them.
- [ ] We need a way for the user to block other users, or even an allowlist for users who can share to them (low prior, only since celebrities may run into an issue of spam if their account email got leaked).

Depends on #78

Chris: P1
Julian: P1
